### ISSUE TITLE
[getter-return]: False positive with abstract property

### ISSUE THREAD

**Issue by jgeurts** [oxc-project/oxc#2349]
Using oxlint v0.2.8, I'm seeing a false positive with eslint(getter-return) using code similar to:

```ts
export abstract class Foo {
  protected abstract get foobar(): number;
}
```

Result is similar to:

```
  ⚠ eslint(getter-return): Expected to always return a value in getter.
    ╭─[src/Foo.ts:2:32]
  1 │ 
  2 │   protected abstract get foobar(): number;
    ·                                ───────────
  3 │ 
    ╰────
  help: Return a value from all code paths in getter.
```
